presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/TestHiveFileSystemAbfs.java

```diff
@@ -0,0 +1,200 @@
+                            .put(EXTERNAL_LOCATION_PROPERTY, getBasePath().toString() + "/" + tableDirectoryName)
+                            .put(BUCKET_COUNT_PROPERTY, 0)
+                            .put(BUCKETED_BY_PROPERTY, ImmutableList.of())
+                            .put(SORTED_BY_PROPERTY, ImmutableList.of())
+                            .build());
+            if (!transaction.getMetadata().listTables(newSession(), Optional.of(table.getSchemaName())).contains(table)) {
+                transaction.getMetadata().createTable(newSession(), tableMetadata, false);
+            }
+            transaction.commit();
+
+            // Hack to work around the metastore not being configured for S3 or other FS.
+            // The metastore tries to validate the location when creating the
+            // table, which fails without explicit configuration for file system.
+            // We work around that by using a dummy location when creating the
+            // table and update it here to the correct location.
+            metastoreClient.updateTableLocation(
+                    database,
+                    table.getTableName(),
+                    getBasePath().toString() + "/" + tableDirectoryName);
+        }
+    }
+
+    private HdfsConfiguration createHdfsConfiguration()
+    {
+        ConfigurationInitializer azureConfig = new PrestoAzureConfigurationInitializer(new HiveAzureConfig()
+                .setAbfsAccessKey(accessKey)
+                .setAbfsStorageAccount(account));
+        return new HiveHdfsConfiguration(new HdfsConfigurationInitializer(new HdfsConfig(), ImmutableSet.of(azureConfig)), ImmutableSet.of());
+    }
+
+    @Override
+    protected Path getBasePath()
+    {
+        return new Path(format("abfs://%s@%s.dfs.core.windows.net/", container, account));
+    }
+
+    @Override
+    @Test
+    public void testRename()
+            throws Exception
+    {
+        Path basePath = new Path(getBasePath(), UUID.randomUUID().toString());
+        FileSystem fs = hdfsEnvironment.getFileSystem(TESTING_CONTEXT, basePath);
+        assertFalse(fs.exists(basePath));
+
+        // create file foo.txt
+        Path path = new Path(basePath, "foo.txt");
+        assertTrue(fs.createNewFile(path));
+        assertTrue(fs.exists(path));
+
+        // rename foo.txt to bar.txt when bar does not exist
+        Path newPath = new Path(basePath, "bar.txt");
+        assertFalse(fs.exists(newPath));
+        assertTrue(fs.rename(path, newPath));
+        assertFalse(fs.exists(path));
+        assertTrue(fs.exists(newPath));
+
+        // rename foo.txt to foo.txt when foo.txt does not exist
+        assertFalse(fs.rename(path, path));
+
+        // create file foo.txt and rename to existing bar.txt
+        assertTrue(fs.createNewFile(path));
+        assertFalse(fs.rename(path, newPath));
+
+        // rename foo.txt to foo.txt when foo.txt exists
+        // This returns true in AzureBlobFileSystem
```

Comment:
This test is long and I'd rather not fork it, since it's not obvious what is different. Let's add an instance check in the base method

```java
// rename foo.txt to foo.txt when foo.txt exists
assertEquals(fs.rename(path, path), !(fs instanceof AzureBlobFileSystem));